query_executor.go

```diff
@@ -33,13 +33,36 @@ func (q *queryExecutor) attemptQuery(qry ExecutableQuery, conn *Conn) *Iter {
 	return iter
 }
 
+// checkErrorIsRetryable evaluates whether a particular error is retryable.
+// In cases that are ambiguous or should be decided on a case-by-case basis it errs on
+// the side of returning true and pushing the decision onto the retry policy.
+func (q *queryExecutor) checkErrorIsRetryable(err error) bool {
```

Comment:
And, +1000

The next logical step would be to add the flag needed to take the idempotence of the query and act accordingly, for instance, if the query is not idempotent, some errors wouldn't be retried; whereas if it's idempotent, then the query would retry. More on that can be found at https://docs.datastax.com/en/developer/java-driver/3.4/manual/retries/